src/connectivity/mod.rs

```diff
@@ -14,6 +14,18 @@ pub trait Connectivity: fmt::Debug {
 
     /// Disconnects from a wireless network currently connected to.
     fn disconnect(&self) -> Result<bool, WifiConnectionError>;
+
+    /// List available ssids
+    fn list_networks(&self) -> Result<Vec<NetworkInfo>, WifiConnectionError>;
```

Comment:
can you annotate this with `#[cfg(target_os = "linux")]` pending when we implement it for other platforms.

unrelated: could you also comb for places in the code were we use `unimplemented!()`, and gate those to only platforms we support. you could also do this is another PR.